src/jsxc.lib.gui.js

```diff
@@ -2644,20 +2644,57 @@ jsxc.gui.window = {
          jsxc.xmpp.sendMessage(message);
       }
 
-      jsxc.gui.window._postMessage(message);
+      /*
+       New Line added to sort the messages
+      */
+      jsxc.gui.window.sortMessages(message);
 
       if (message.direction === 'out' && message.msg === '?' && jsxc.options.get('theAnswerToAnything') !== false) {
          if (typeof jsxc.options.get('theAnswerToAnything') === 'undefined' || (Math.random() * 100 % 42) < 1) {
-            jsxc.options.set('theAnswerToAnything', true);
+	         jsxc.options.set('theAnswerToAnything', true);
 
-            jsxc.gui.window.postMessage(new jsxc.Message({
-               bid: message.bid,
-               direction: jsxc.Message.SYS,
-               msg: '42'
-            }));
+	         jsxc.gui.window.postMessage(new jsxc.Message({
+	            bid: message.bid,
+	            direction: jsxc.Message.SYS,
+	            msg: '42'
+	         }));
          }
       }
+      return message;
+   },
+
+   sortMessages: function(message){
+      var messageArray = jsxc.options.allMessages;
+      var length = messageArray.length;
+      var position = length;
+      var firstMessage;
 
+      do{
+         if (typeof messageArray[position-1] === 'undefined') {
+            firstMessage = true;
+         }
+         if (isNaN(message.stamp)) {
+            length = messageArray.length;
```

Comment:
I mean, length was defined a couple of lines above and I think it is never changed.